include/sys/ring_buffer.h

```diff
@@ -173,7 +173,7 @@ __deprecated static inline void sys_ring_buf_init(struct ring_buf *buf,
  *
  *  @return Ring buffer free space (in 32-bit words or bytes).
  */
-static inline int z_ring_buf_custom_space_get(u32_t size, u32_t head,
+static inline u32_t z_ring_buf_custom_space_get(u32_t size, u32_t head,
```

Comment:
can you have these return size_t instead?

Reply:
I agree that size_t would be more appropriate for the return types, but u32_t what is used as a size type for the ring_buf structure head, tail and size.

As such, it is also the type the rest of the ring_buf API uses (with the exception of the functions fixed in this PR). For example:

```c
/**
 * @param[in]  size Requested size (in bytes).
 *
 * @return Number of valid bytes in the provided buffer which can be smaller
 *	   than requested if there is not enough free space or buffer wraps.
 */
u32_t ring_buf_get_claim(struct ring_buf *buf, u8_t **data, u32_t size);
```

So even if size_t is more appropriate, if it was to be used it should be used as the size type for the ring_buf structure and the entire ring_buf API. This wouldn't be such a minor change (involves changing ring_buffer itself and the API calls throughout the code).

Having said that, I wouldn't be unwilling to make a complete change of the API if you guys are okay with it.

Reply:
gotcha. I think this can go in as you have written it, but of course if you want to overhaul the API to use a better set of types I'd love to review it.